emily/handler/src/database/entries/deposit.rs

```diff
@@ -122,25 +126,37 @@ impl DepositEntry {
                 format!("last update block height is inconsistent between history and top level data. {stringy_self:?}")
             ));
         }
-        if self.status != latest_event.status {
+        if self.status != (&latest_event.status).into() {
             return Err(Error::Debug(
                 format!("most recent status is inconsistent between history and top level data. {stringy_self:?}")
             ));
         }
         Ok(())
     }
+
+    /// Gets the latest event.
+    pub fn latest_event(&self) -> &DepositEvent {
+        self.history
+            .last()
+            .expect("Deposit entry must always have at least one event.")
+    }
```

Comment:
I would update the comment to note that this entry needs to be validated before calling this function.

Reply:
Made this return a result instead.